scrapper.py

```diff
@@ -25,39 +34,117 @@ class Crawler:
     """
     Crawler implementation
     """
+
     def __init__(self, seed_urls, max_articles: int):
-        pass
+        self.seed_urls = seed_urls
+        self.max_articles = max_articles
+        self.urls = []
 
     def _extract_url(self, article_bs):
-        pass
+        self.urls.append('http://www.elista.org' + article_bs.find('a')['href'])
 
     def find_articles(self):
         """
         Finds articles
         """
-        pass
+        for url in self.seed_urls:
+            response = requests.get(url)
+            response.encoding = 'utf-8'
+            page_bs = BeautifulSoup(response.text, features='html.parser')
+            class_bs = page_bs.find_all('div', class_='grid_19 omega')
+            for article_bs in class_bs:
+                if len(self.urls) < self.max_articles:
+                    self._extract_url(article_bs)
 
     def get_search_urls(self):
         """
         Returns seed_urls param
         """
-        pass
+        return self.seed_urls
+
+
+class HTMLParser:
+    def __init__(self, article_url, article_id):
+        self.article_url = article_url
+        self.article_id = article_id
+        self.article = Article(article_url, article_id)
+
+    def _fill_article_with_text(self, article_bs):
+        """
+        fills article instance with text of article
+        """
+        text_bs = article_bs.find('div', class_='fullnews_content')
+        self.article.text = text_bs.text
+
+    def _fill_article_with_meta_information(self, article_bs):
+        """
+        fills article instance with title, date and topics
+        """
+        meta_bs = article_bs.find('div', class_='news_item fullnews')
+        title_bs = meta_bs.find('h1').text
+        self.article.title = title_bs
+
+        date_bs = meta_bs.find('span', class_='date').text
+        date = datetime.strptime(date_bs, '%d.%m.%Y : %H.%M')
+        self.article.date = date
+
+        topic = article_bs.find('h4', class_='chapter_h').text
+        self.article.topics.append(topic)
+
+        self.article.author = 'NOT FOUND'
+
+    def parse(self):
+        """
+        does all the work
+        """
+        response = requests.get(self.article_url)
```

Comment:
no headers or other settings. Can you collect with such setup 100 articles safely?

Reply:
yes, elista.org has no DDOS defense